Sources/Fuzzilli/Core/ProgramBuilder.swift

```diff
@@ -1317,28 +1317,28 @@ public class ProgramBuilder {
 
     public struct SwitchBuilder {
         public typealias SwitchCaseGenerator = () -> ()
-        fileprivate var defaultCaseGenerator: SwitchCaseGenerator? = nil
-        fileprivate var caseGenerators: [(value: Variable, fallsthrough: Bool, body: SwitchCaseGenerator)] = []
+        fileprivate var firstCaseGenerator: SwitchCaseGenerator? = nil
+        fileprivate var caseGenerators: [(value: [Variable], fallsthrough: Bool, body: SwitchCaseGenerator)] = []
 
-        public mutating func addDefault(_ generator: @escaping SwitchCaseGenerator) {
-            assert(defaultCaseGenerator == nil)
-            defaultCaseGenerator = generator
+        public mutating func addFirstCase(_ generator: @escaping SwitchCaseGenerator) {
```

Comment:
Do we still need this at all? It feels a bit weird, and I guess even an empty Switch statement (`switch (x) { }`) is valid JavaScript, so I don't think we need to handle the first case differently from the others. How about we do it like this:
```swift
public mutating func addDefault(fallsThroughToFollowingCase fallsThrough: Bool = false, body: @escaping SwitchCaseGenerator) {
    caseGenerators.append((nil, fallsThrough, body))
}

public mutating func add(_ v: Variable, fallsThroughToFollowingCase fallsThrough: Bool = false, body: @escaping SwitchCaseGenerator) {
    caseGenerators.append((v, fallsThrough, body))
}
```

That should make for a fairly good API.

Then it's also obvious that nothing can fallthrough to the first case, and whether the last case fallsthrough or not doesn't matter. Now, the `doSwitch` method just needs to do a bit more work by tracking whether the _previous_ case falls through when emitting the current case, but I think overall it will be simpler and more intuitive.

Reply:
I've made these changes to `SwitchBuilder` and updated `doSwitch` to handle the first case. Not entirely happy with the way it's currently written but I can't think of a simpler alternative. Could you give it a look and let me know your thoughts?